spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/neo4j/DriverConfiguration.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2012-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.autoconfigure.neo4j;
+
+import org.neo4j.driver.AuthToken;
+import org.neo4j.driver.Config;
+import org.neo4j.driver.Driver;
+import org.neo4j.driver.GraphDatabase;
+
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * Creates an instance of the Neo4j Java driver.
+ *
+ * @author Michael J. Simons
+ */
+@Configuration(proxyBeanMethods = false)
+final class DriverConfiguration {
+
+	@Bean
+	@ConditionalOnMissingBean(Driver.class)
+	Driver neo4jDriver(final Neo4jDriverProperties driverProperties) {
+
+		final AuthToken authToken = driverProperties.getAuthentication().asAuthToken();
+		final Config config = driverProperties.asDriverConfig();
+
+		return GraphDatabase.driver(driverProperties.getUri(), authToken, config);
```

Comment:
There must be a fallback for `spring.data.neo4j.uri` and a metadata with warning deprecation should indicate the replacement.

Reply:
There is no plan for a deprecation phase for the (soon) former SDN properties.
The SDN 6 replacement will be so disruptive that it just does not make any sense to respect those values anymore. The SDN starter will require this auto-configuration to reflect the truth for those properties: They belong into the driver and not into the higher level configuration.
In this hopefully short period of having this change merged and the pending SDN PR merge, I don't really consider to support the former properties.

Reply:
> There is no plan for a deprecation phase for the (soon) former SDN properties.

I get that but that property is in Spring Boot and part of its public API so I was hoping we could honour it still if that makes any sense. I am thinking of the use case of someone using Neo4j with a custom URI and/or username/password authentication. 

Is there any problem to honour the existing property in a deprecated fashion?

Reply:
SDN will not know about those properties in the future because it does not configure the `Driver` anymore. The only place I see this is in the driver configuration but this would mix up the responsibilities for the `spring.data.neo4j` namespace.
If there is a away to alias properties, this would be helpful. So, I could say <legacy-property> is deprecated but gives it value to <new-property> Unfortunately I am not aware of any.
I am aware that a sudden drop without deprecation is not the finest thing to do but yet it seems like the cleanest solution. Otherwise we would have to deprecate everything (also in SDN 5.3.x) and the users only see striked through text everywhere.

Reply:
To be clear, I am not talking about SDN at all. This is purely a Spring Boot topic. This is 100% on the user side too. The use case I have in mind is someone having `spring.data.neo4j.uri=xxxx` in their configuration and upgrading to 2.4. Rather than trying to connect to a Neo4j instance on localhost, I would like the auto-configuration of the driver to pick up this property as a fallback. 

> If there is a away to alias properties

Yes, there is. This is what I am suggesting here. I just wanted to make sure this wasn't a bad idea before trying to give it a try. I'll do that when polishing this one but feel free to comment if, based on this additional context, you think we shouldn't.

Reply:
We are definitely on the same page here. I just wanted to underline that SDN will not be as it was before from an API perspective. That was my reasoning for "just dropping" the old properties.
I think it is a welcoming transition help for <2.4 users going for 2.4 but should be deprecated on arrival 😺 
If you can give me a hint how to make the old properties an alias for the new ones, I could try this myself. My mind somehow blocks the idea of throwing (partial) ownership for `spring.data.neo4j` and `spring.neo4j` in the same place and the rest of the real new `spring.data.neo4j` properties will get handled by another auto-configurer.

Reply:
Alright. I can give that a try as part of polishing this next week.